### PR TITLE
keras_api: expose noise multiplier getter

### LINKED ISSUE #26: Provide a way to get the calculated noise multiplier through public Keras API
Feed back from a user:

"""
It seems like when i don't set a specific noise multiplier in the DP_Keras_Config and instead let the API determine one for the given epsilon budget the field in the original DP_Keras_Config object is no correctly updated.
More concretely when i do:
```
params = keras_api.DPKerasConfig(
                epsilon=FLAGS.epsilon,
                delta=5.1-05,
                clipping_norm=10.0,
                batch_size=1024,
                gradient_accumulation_steps=1,
                train_steps=1900,
                train_size=19601,
                seed=42,
            )
model = keras_api.make_private(model, params)
```
I get the following correct print statement: 

"Finished calculating noise multiplier: calculated_noise_multiplier=0.4813812471830989"

But then:

```
print(params.noise_mutiplier)
```

stil yields "None".
For logging purposes it would be great to expose the DPkeras config with the updated field to the user i think :thinking_face: ...
"""

`params` are not updated because it is immutable object. It should probably stay so. We should provide a way to get the noise_multiplier via public api.

### PR DESCRIPTION
**Pull request by staticpayload**
## Summary
- expose a public helper and model method to read the configured or calibrated noise multiplier
- cache the calibrated value on the model and reuse it in DP-SGD gradient noise
- add tests for config-based and calibrated retrieval

## Testing
- pytest tests/keras_api_test.py (Amazon Linux 2023, Python 3.11, eu-central-2)

Fixes #26

### CODE REVIEW

File: jax_privacy/keras_api.py [hunk 1]
```diff
@@ -256,6 +257,41 @@ def make_private(model: keras.Model, params: DPKerasConfig) -> keras.Model:
...
 
+def get_noise_multiplier(model: keras.Model) -> float:
+  """Returns the noise multiplier used for DP-SGD training.
+
+  If the noise multiplier is not set in DPKerasConfig, this will calibrate it
+  once and cache the value on the model.
+
+  Example:
+    >>> import keras  # doctest: +SKIP
+    >>> from jax_privacy import keras_api  # doctest: +SKIP
```
Comment (ryan112358) [inline jax_privacy/keras_api.py]: Why are these doctests skipped?

I'm not sure this doctest belongs on this method,  I think it better fits in the module-level docstring

**Review by ryan112358 — CHANGES_REQUESTED** [review]

File: jax_privacy/keras_api.py [hunk 2]
```diff
@@ -256,6 +281,22 @@ def make_private(model: keras.Model, params: DPKerasConfig) -> keras.Model:
   return model
 
 
+def get_noise_multiplier(model: keras.Model) -> float:
+  """Returns the noise multiplier used for DP-SGD training.
+
+  If the noise multiplier is not set in DPKerasConfig, this will calibrate it
```
Comment (ryan112358) [inline jax_privacy/keras_api.py:292]: pylint is complaining about the docstring format.  Valid formats include:

a) one-lin edocstring (delete this extra stuff)
b) full docstring, but it needs to include an "Arguments" and "Returns" block

File: jax_privacy/keras_api.py [hunk 3]
```diff
@@ -533,11 +576,29 @@ def _dp_train_step(
...
+    model._dp_noise_multiplier = noise_multiplier  # pylint: disable=protected-access
+  return noise_multiplier
+
+
 def _noised_clipped_grads(
     compute_loss_and_updates_fn: LossFn,
     dp_params: DPKerasConfig,
     state: _StateType,
     data: _KerasInputsDataType,
+    model: keras.Model | None = None,
```
Comment (ryan112358) [inline jax_privacy/keras_api.py:621]: Add to docstring

Comment (ryan112358) [reply jax_privacy/keras_api.py:621]: I think this is the main thing blocking submission.  Here is the error I see:

Args section is missing, does not match arg list or is badly indented.

Expected: compute_loss_and_updates_fn, dp_params, state, data, model

Found   : compute_loss_and_updates_fn, dp_params, state, data

Missing: model

**Review by ryan112358 — APPROVED** [review]
